packages/cactus-plugin-keychain-memory/src/main/typescript/prometheus-exporter/prometheus-exporter.ts

```diff
@@ -0,0 +1,49 @@
+import promClient from "prom-client";
+import { Keys } from "./response.type";
+import { totalKeyCount } from "./metrics";
+
+export const K_CACTUS_KEYCHAIN_MEMORY_TOTAL_KEY_COUNT =
+  "cactus_keychain_memory_total_key_count";
+
+export interface IPrometheusExporterOptions {
+  pollingIntervalInMin?: number;
+}
+
+export class PrometheusExporter {
+  public readonly metricsPollingIntervalInMin: number;
+  public readonly keys: Keys = { counter: 0 };
+
+  constructor(
+    public readonly prometheusExporterOptions: IPrometheusExporterOptions,
+  ) {
+    this.metricsPollingIntervalInMin =
+      prometheusExporterOptions.pollingIntervalInMin || 1;
+  }
+
+  public addCurrentKey(): void {
+    this.keys.counter++;
+    totalKeyCount
+      .labels(K_CACTUS_KEYCHAIN_MEMORY_TOTAL_KEY_COUNT)
+      .set(this.keys.counter);
+  }
+
+  public removeCurrentKey(): void {
+    this.keys.counter--;
```

Comment:
@jagpreetsinghsasan I think this is redundant with the map itself holding it's own keycount in the `size` property as explained in one of my above comments.
@see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/size